include/nrf_rpc/nrf_rpc_ipc.h

```diff
@@ -57,7 +68,7 @@ struct nrf_rpc_ipc {
 	void *context;
 
 	/** Indicates if transport is already initialized. */
-	bool used;
+	enum nrf_rpc_ipc_state state;
```

Comment:
nit: I suggest avoiding enumeration in header files because using different compiler options for building a library exposing this header than the code using the library, the size of this structure field could differ.

How about `typedef uint8_t nrf_rpc_ipc_state;` to avoid this potential issue?
```suggestion
	nrf_rpc_ipc_state state;
```

Reply:
I forgot to add this comment. I have strong believe that `init` and `sent` can be called from different context. Doesn't we need to make state atomic?

Reply:
We do not need the typedef here. I can put uint8_t directly into the structure. It is used internally only.
Moreover - with that change I can move enumerations into *.c file.

Reply:
Done

Reply:
Discussed offline

Reply:
> I forgot to add this comment. I have strong believe that `init` and `sent` can be called from different context. Doesn't we need to make state atomic?

For the time being and how it is implemented right now we do not need synchronization. The initialization looks only if the state is UNINITIALIZED. If it is not the function would exit. In the same time if the state is UNINITIALIZED the send function would exit immediately. The place where the state is changed in initialization function it taken carefully to be changed only when we are ready for the send function to wait. It means that even if the send function would be called before init function leaves but it already changed the state - the further processing would be as expected.

I suppose the special care would be needed when we would allow the communication to reinitialize. But lets keep it simple if we do not need it right now.